File: frontend/src/types/product.ts
// backend/app/schemas/product.py::ProductRead ile birebir eşleşir.
export interface ProductRead {
  id: number;
  name: string;
  sku: string;
  category?: string | null;
  default_price: number;
  cost_price?: number | null;
  best_before_date?: string | null;
  is_active: boolean;
}

export interface ProductListOut {
  items: ProductRead[];
  total: number;
}

export interface ProductCreatePayload {
  name: string;
  sku: string;
  category?: string | null;
  default_price: number;
  cost_price?: number | null;
  best_before_date?: string | null;
}

export type ProductUpdatePayload = Partial<ProductCreatePayload>;

// backend/app/schemas/product_import.py::ImportRowErrorOut ile birebir eşleşir.
export interface ImportRowError {
  row: number | null;
  message: string;
}

// backend/app/schemas/product_import.py::ImportResultOut ile birebir eşleşir.
export interface ImportResult {
  created: number;
}

File: backend/app/schemas/product.py
from datetime import date

from pydantic import BaseModel, ConfigDict


class ProductBase(BaseModel):
    name: str
    sku: str
    category: str | None = None
    default_price: float
    cost_price: float | None = None
    best_before_date: date | None = None


class ProductCreate(ProductBase):
    pass


class ProductUpdate(BaseModel):
    name: str | None = None
    sku: str | None = None
    category: str | None = None
    default_price: float | None = None
    cost_price: float | None = None
    best_before_date: date | None = None


class ProductRead(ProductBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    is_active: bool


class ProductListOut(BaseModel):
    items: list[ProductRead]
    total: int
